src/language/concurrency.md

```diff
@@ -144,80 +171,107 @@ void main() [!async!] {
 Once `readAsString()` returns a value, Dart code execution resumes.
 
-![Flowchart-like figure showing app code executing from start to exit, waiting for native I/O in between](/assets/img/language/concurrency/basics-await.png)
+![Flowchart-like figure showing app code executing from start to exit, waiting
+for native I/O in between](/assets/img/language/concurrency/basics-await.png)
 
-If you'd like to learn more about using `async`, `await`, and futures,
-visit the [asynchronous programming codelab][].
+### Streams
 
-[asynchronous programming codelab]: /codelabs/async-await
+Dart also supports asynchronous code in the form of streams. Streams
+provide values in the future and repeatedly over time. A promise to provide a
+series of `int` values over time has the type `Stream<int>`.
 
+In the following example, the stream created with `Stream.periodic`
+repeatedly emits a new `int` value every second.
 
-## How isolates work
+<?code-excerpt "lib/stream_syntax.dart"?>
+```dart
+Stream<int> stream = Stream.periodic(const Duration(seconds: 1), (i) => i * i);
+```
+
+#### await-for and yield
+
+Await-for is a type of for loop that executes each subsequent iteration of the
+loop as new values are provided. In other words, it’s used to “loop over”
+streams. In this example, a new value will be emitted from the function
+`sumStream` as new values are emitted from the stream that’s provided as an
+argument. The `yield` keyword is used rather than `return` in functions that 
+return streams of values.
+
+<?code-excerpt "lib/await_for_syntax.dart"?>
+```dart
+Stream<int> sumStream(Stream<int> stream) async* {
+  var sum = 0;
+  await for (final value in stream) {
+    yield sum += value;
+  }
+}
+```
+
+If you'd like to learn more about using `async`, `await`, `Stream`s and
+`Future`s,  visit the [asynchronous programming codelab][].
+
+[asynchronous programming codelab]: /codelabs/async-await
 
-Most modern devices have multi-core CPUs.
-To take advantage of multiple cores,
-developers sometimes use shared-memory threads running concurrently.
-However, shared-state concurrency is
-[error prone](https://en.wikipedia.org/wiki/Race_condition#In_software) and
-can lead to complicated code.
+## Isolates
 
-Instead of threads, all Dart code runs inside of isolates.
-Each isolate has its own memory heap,
-ensuring that none of the state in an isolate is accessible from
-any other isolate.
-No shared state between isolates means concurrency complexities like 
+Dart also supports concurrency via isolates. Most modern devices have multi-core
+CPUs. To take advantage of multiple cores, developers sometimes use
+shared-memory threads running concurrently. However, shared-state concurrency is
+[error prone](https://en.wikipedia.org/wiki/Race_condition#In_software) and can
+lead to complicated code.
+
+Instead of threads, all Dart code runs inside isolates. Each isolate has its
+own global fields, ensuring that none of the state in an isolate is accessible
```

Comment:
The last paragraph seems like it fits better up here (deleted in subsequent comment):

```suggestion
Instead of threads, all Dart code runs inside isolates.
Using isolates, your Dart code can perform multiple independent tasks at once.
Isolates are like threads or processes, but each isolate has its own memory
and a single thread running an event loop.

Each isolate has its own global fields,
ensuring that none of the state in an isolate is accessible
```

Notice I removed the clause "using additional processor cores if they're available." While that might be the case, I think it introduces unnecessary uncertainty, like "do I have additional processor cores available? what if I don't, can I use isolates? what happens if I try?" and stuff like that. If there's nothing the user can do about it and we're not going to elaborate on what to do otherwise, probably best not to mention it.

If you think it's the kind of thing that's important for developers to know or that they'd want to know, add it back (I don't have that context)